### ISSUE TITLE
go c id / app can teleport you to a place where you fall under floor

### ISSUE THREAD

**Issue by Aokromes** [TrinityCore/TrinityCore#30528]
### Description

when testing https://github.com/TrinityCore/TrinityCore/issues/30497#issuecomment-2542016918 i found that go c id and app can teleport you to a place where you fall under floor

### Expected behaviour

teleport you to land level.

### Steps to reproduce the problem

go c id 2830
fall

### Branch

3.3.5

### TC rev. hash/commit

72f148916c1b

### Operating system

debian 12 x64

### Custom changes

n/r

**Comment by CraftedRO** [comment]
I think here there's need a decision about command functionallity, at the moment if a npc have undergound spawn, this command will teleport you underground even if in some cases that npc is properly ajusted to be on map see .go c 25623 example

**Comment by CraftedRO** [comment]
Imo is good as it is, to be teleported exactly where that npc have spawn coordinates (Z), as a gm you also have fly so I don't think  would be an issue if you "fall".

**Comment by Shauren** [comment]
I agree, going to EXACTLY the same coords is better for reproducing issues

**Comment by Aokromes** [comment]
<img width="1922" height="1200" alt="Image" src="https://github.com/user-attachments/assets/3ce6656f-a01b-48b8-9ab3-6cac7694ec82" />

...

**Comment by CraftedRO** [comment]
> ...

As I said the command is good as it is and .go c 7614 spawn position (Z) need ajustments
